File: JavaScript/LeetCode75/Q1732-FindTheHighestAltitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function(gain) {
    gain.unshift(0);
    for(let i = 1; i < gain.length; i++) {
        gain[i] = gain[i] + gain[i-1];
    }
    return Math.max(...gain);
};

const largestAltitude = gain => {
    let prev = 0;
    let big = 0;
    
    for(let i = 0; i < gain.length; i++) {
        prev += gain[i];
        if(prev > big) big = prev;
    }
    return big;
}

File: Python/Leetcode75/Easy/PrefixSum/Q1732-FindTheHighestAltitude.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        n = len(gain)

        altitudes = [0] * (n + 1)
        for i in range(len(gain)):
            altitudes[i + 1] = altitudes[i] + gain[i]
        
        return max(altitudes)